src/audio/rtnr/rtnr.c

```diff
@@ -214,76 +220,73 @@ static int32_t rtnr_check_config_validity(struct comp_dev *dev,
 		return -EINVAL;
 	}
 
-	rtnr_set_process_sample_rate(dev, cd->config.params.sample_rate);
+	rtnr_set_process_sample_rate(mod, cd->config.params.sample_rate);
 
 	return 0;
 }
 
-static struct comp_dev *rtnr_new(const struct comp_driver *drv,
-				 const struct comp_ipc_config *config,
-				 const void *spec)
+static int rtnr_init(struct processing_module *mod)
 {
-	const struct ipc_config_process *ipc_rtnr = spec;
-	struct comp_dev *dev;
+//const struct comp_driver *drv, const struct comp_ipc_config *config, const void *spec
+
+	struct module_data *mod_data = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	struct module_config *ipc_rtnr = &mod_data->cfg;
 	struct comp_data *cd;
 	size_t bs = ipc_rtnr->size;
 	int ret;
 
-	comp_cl_info(&comp_rtnr, "rtnr_new()");
+	comp_info(dev, "rtnr_new()");
 
 	/* Check first before proceeding with dev and cd that coefficients
 	 * blob size is sane.
 	 */
 	if (bs > SOF_RTNR_MAX_SIZE) {
-		comp_cl_err(&comp_rtnr, "rtnr_new(), error: configuration blob size = %u > %d",
-					bs, SOF_RTNR_MAX_SIZE);
-		return NULL;
+		comp_err(dev, "rtnr_new(), error: configuration blob size = %u > %d",
+			 bs, SOF_RTNR_MAX_SIZE);
+		return -EINVAL;
 	}
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-
-	dev->ipc_config = *config;
-
 	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
 	if (!cd)
-		goto err;
+		return -ENOMEM;
 
-	comp_set_drvdata(dev, cd);
+	mod_data->private = cd;
 
 	cd->process_enable = true;
 
 	/* Handler for component data */
 	cd->model_handler = comp_data_blob_handler_new(dev);
 	if (!cd->model_handler) {
-		comp_cl_err(&comp_rtnr, "rtnr_new(): comp_data_blob_handler_new() failed.");
+		comp_err(dev, "rtnr_new(): comp_data_blob_handler_new() failed.");
+		ret = -ENOMEM;
 		goto cd_fail;
 	}
 
 	/* Get initial configuration from topology */
-	ret = rtnr_set_config_bytes(dev, ipc_rtnr->data, ipc_rtnr->size);
```

Comment:
@MingJenTai Can you please help with this. Has this worked, I wonder a current kernel provides the configuration blob in comp new() IPC?